ecal/core/src/io/ecal_memfile.h

```diff
@@ -161,12 +163,31 @@ namespace eCAL
     bool HasReadAccess()     const {return(m_access_state == access_state::read_access);};
     bool HasWriteAccess()    const {return(m_access_state == access_state::write_access);};
 
+    
+    // @deprecate_eCAL6
+    // Use of platform specific aligment to remain compatible with previous struct layout
+    // This can be harmonized for all platforms in a later version of eCAL that drops compatibility
+#pragma pack(push, 1)
     struct SInternalHeader
     {
-      unsigned short int_hdr_size  = sizeof(SInternalHeader);
-      unsigned long  cur_data_size = 0;
-      unsigned long  max_data_size = 0;
+      static_assert(sizeof(std::array<std::uint8_t, 1>) == 1, "Memory layout of std::array is different from C-style array.");
+      
+      std::uint16_t int_hdr_size = sizeof(SInternalHeader);
+#if _WIN32 || _WIN64 || (INTPTR_MAX == INT32_MAX)    // Use of standard 32 bit data types on windows and other 32 bit platforms to remain compatible with built-in "long" data type
+                                                     // of previous struct layout. For some reason 64-bit msvc uses 4 bytes alignment as well.
+      std::array<std::uint8_t, 2> _reserved_0;       // Add 2 bytes padding for 4 bytes alignment on Windows  
```

Comment:
Those should be initialized to 0 as well. While this is not relevant now, this may be relevant when we later use one of those fields. Then the legacy applications should always send a 0 there to not confuse newer applications that read an actual value from those fields.

```suggestion
      std::array<std::uint8_t, 2> _reserved_0 = {};       // Add 2 bytes padding for 4 bytes alignment on Windows  
```

Reply:
In case of _reserved_0 i don't think that it is necessary to zero the memory, since the field will never be used for future applications and is only intended to be compatible to the padding of the previous header layout (which maybe already contains non-zero values).

In the next version of the header I will introduce an additional version field to query which fields or rather features are available for the memory file. Then it can be easily extended for future use cases.

Reply:
Hm, OK, but it still feels wrong (to me) to write uninitialized memory to a file. But should be OK, if you prefer it like that.

Reply:
I can change it if you feel better with zeros in the reserved field. The overhead of initialization should be acceptable. ;-)